### ISSUE TITLE
Meteor 1.3 module import failing

### ISSUE THREAD

**Issue by talha-asad** [meteor/meteor#7420]
I can't import [this](https://www.npmjs.com/package/facebook-ads-sdk) npm package. I get an error thrown like this: 
```bash
W20160717-02:48:31.047(5)? (STDERR) import Http from './http'
W20160717-02:48:31.047(5)? (STDERR) ^^^^^^    
W20160717-02:48:31.049(5)? (STDERR) SyntaxError: Unexpected reserved word
```

I am trying to import it like below, the module recommends to be imported that way as well.
```js
import { FacebookAdsApi } from 'facebook-ads-sdk/src/api';
```

**Comment by laosb** [comment]
When you try to `import { FacebookAdsApi } from 'facebook-ads-sdk/src/api'`, you're actually importing [this file](https://github.com/lucascosta/facebook-js-ads-sdk/blob/master/src/api.js), which is in ES6. Meteor doesn't do transpile on files in npm package, so this error occured.

I think by saying `import FacebookAdsApi from './[...]/src/api'` in their README, they mean placing all files in this package manually to your application and then import. Btw, you're adding a pair of extra brackets surrounding the variable you imported. That's not intended as that file only have a default export.

Closing.

**Comment by talha-asad** [comment]
No they don't mean manually placing these files in your application. Meteor should be able to import ES6 files from node modules, otherwise whats the use of writing in ES6 if already written ES6 code is not supported. This is a bug and should be marked as such. Yes and that file is exactly what I am trying to import.

**Comment by abernix** [comment]
@talha-asad You're absolutely right that they don't mean manually placing the files in your application.

As you said, writing in ES6 does no use if the code is not supported.  But this is why transpilation exists: to provide the ability to write in ES6 in a world that isn't quite ready for it.  NPM packages are meant to run in a `node` project and they do this without additional transpilation beyond the state that the package author shipped the package in.  Meteor is no different in this regard.

For this reason, NPM package maintainers should be choosing what Node platforms they want to target and transpiling their code to run on those platforms.  This should be specified in the package description and [in the `package.json`](https://docs.npmjs.com/files/package.json#engines).  The alternative requires double-transpilation of packages and a significant amount of overhead being added to the process, especially with the extensive dependency trees in some NPM packages.

Obviously, on certain platforms with basic ES6 functions (`const`, `let`, etc.), transpilation wouldn't be necessary and with various `harmony` flags certain things can be enabled natively, but these are experimental until made official.  I think you'll see this problem greatly improve in Meteor 1.4 (which ships with Node 4.4.7) and [supports a lot more ES6 natively](http://node.green/) but it's unlikely to go away entirely until all ECMAscript is supported by all platforms (something that's not likely to happen for a while).

**tl;dr: NPM packages should be transpiled _once_ by their authors for the platforms they expect to run on**

Luckily, in your case @talha-asad, `facebook-ads-sdk` _does_ transpile its code (as it should) and you simply need to:

    import { FacebookAdsApi } from 'facebook-ads-sdk';

(instead of importing directly from the raw `/src/` directory)

**Comment by thamizharasan** [comment]
Yes. I simulated and it actually works. Facebook sdk is working. Thanks.

**Comment by talha-asad** [comment]
`FacebookAdsApi` can be imported that way, but what about other objects? I am already using this library by using the good old `require` so usage isn't an issue. What I meant to report here was that Meteor doesn't needs to be extra clever when it comes to importing from a file in an NPM package, especially if it's ES6 compliant, because it needs *no* transpilation and Errors like:

```bash
W20160717-02:48:31.047(5)? (STDERR) import Http from './http'
W20160717-02:48:31.047(5)? (STDERR) ^^^^^^    
W20160717-02:48:31.049(5)? (STDERR) SyntaxError: Unexpected reserved word
```
are totally misleading. I wonder if this is how it is meant to be designed.

cc @benjamn

**Comment by abernix** [comment]
> what about other objects?

I'm not sure what you mean.  This is true for most NPM packages.

> I am already using this library by using the good old require so usage isn't an issue

You don't need to, you can use `import`!

> Meteor doesn't needs to be extra clever when it comes to importing from a file in an NPM package, especially if it's ES6 compliant, because it needs no transpilation

The `import` syntax as you used it (and `facebook-ads-sdk/src/api` module that you were imported) **do** require transpilation.  The error you're seeing is from v8, not from Meteor.

**Comment by talha-asad** [comment]
```js
import FacebookAdsApi from './[...]/src/api'
import { AdAccount, Campaign } from './[...]/src/objects'
```
`AdAccount`, `Campaign`, etc. These objects.

> The error you're seeing is from v8, not from Meteor.
If this is true than, maybe v1.4 should be released with Node v6 instead of v4. Is v8 running with all the harmony flags? I think 0.10.x with all flags does support import statements. Will check on this.

**Comment by abernix** [comment]
Anything that is enabled with a harmony flag is not considered stable and `import`/`export` haven't even officially landed in v8 yet.

> `AdAccount`, `Campaign`, etc. These objects.

There's no difference.  They are imported exactly the same as I suggested before:

```js
import { FacebookAdsApi, AdAccount, Campaign } from 'facebook-ads-sdk';
```

See the transpiled source of the NPM module you're using and check out the exports:
https://github.com/lucascosta/facebook-js-ads-sdk/blob/master/dist/cjs.js#L2225-L2237

**Comment by talha-asad** [comment]
Thank you for the details, yes that is how require, works right now as well.

Will this remain the same way in Meteor 1.4 with Node v4 or will I be able to use the way mentioned in the ReadMe then?

**Comment by laosb** [comment]
Meteor won't change this behavior, I think. However with higher Node version, this problem can be eased by the native support of ES6 features.

**Comment by abernix** [comment]
The way that package author has defined things in the README will not work until v8 supports modules natively or without enabling Harmony flags.  Meteor 1.4 will not change this.  I think it's reasonably to use the syntax I've recommended until then.  You could pass your own harmony flags to `node` with `NODE_OPTIONS` but this will not be supported and is untested.
